src/api.ts

```diff
@@ -84,14 +84,14 @@ export const apiFactory = (serviceId: number) => {
 		metrics.inc(Metrics.OnlineDevices);
 		metrics.inc(Metrics.TotalDevices);
 
-		const workerId = req.params.worker;
+		const workerId = parseInt(req.params.worker, 10);
 		const uuid = req.body.common_name;
-		if (workerMap[uuid] != null && workerMap[uuid] !== req.params.worker) {
+		if (workerMap[uuid] != null && workerMap[uuid] !== workerId) {
 			metrics.dec(Metrics.OnlineDevices);
 		}
 
 		workerMap[uuid] = workerId;
-		clients.setConnected(uuid, serviceId, true, logger);
+		clients.setConnected(uuid, serviceId, workerId, true, logger);
```

Comment:
Can we add a log statement here that logs uuid, serviceId & workerId? Will be useful for debugging these kind of issues.